### PR TITLE
Unit Test Implementation

### PR DESCRIPTION
**Pull request by orhungorkem**
I implemented unit tests for creating posts.

### CODE REVIEW

File: practiceapp/src/test/java/com/practiceapp/practiceapp/service/PostServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        postEntity.setText("Mock post");
+    }
+
+
+    @Test
+    void savePost() {
+
+        when(postRepository.save(any(PostEntity.class))).thenReturn(postEntity);
+
+        int response = postService.savePost(postEntity);
```
Comment (bekirkeldal) [inline practiceapp/src/test/java/com/practiceapp/practiceapp/service/PostServiceTest.java:42]: A comment might be added to indicate that this function call returns 1 when everything is OK.

**Review by bekirkeldal — COMMENTED** [review]
A comment suggestion is provided.

File: practiceapp/src/test/java/com/practiceapp/practiceapp/controller/post/CreatePostControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    public void savePostTest() throws Exception {
+
+
+
+        PostService postService = mock(PostService.class);  //mock the post service
+        PostRepository postRepository = mock(PostRepository.class);   //mock post repository
+
+        //mock the behavior of service entity and repository
+        when(postService.savePost(any(PostEntity.class))).thenReturn(1);
+        when(postRepository.save(any(PostEntity.class))).thenReturn(mock_post);
```
Comment (irem-zeynep) [inline practiceapp/src/test/java/com/practiceapp/practiceapp/controller/post/CreatePostControllerTest.java:66]: The controller communicates with the service, not the repository.  So, I'm not sure mocking repo is necessary.

**Review by irem-zeynep — COMMENTED** [review]
There may be no need to mock the repository in the controller.

File: practiceapp/src/test/java/com/practiceapp/practiceapp/service/PostServiceTest.java (hunk 1 above)
Comment (orhungorkem) [reply practiceapp/src/test/java/com/practiceapp/practiceapp/service/PostServiceTest.java:42]: I am adding it.

File: practiceapp/src/test/java/com/practiceapp/practiceapp/controller/post/CreatePostControllerTest.java (hunk 2 above)
Comment (orhungorkem) [reply practiceapp/src/test/java/com/practiceapp/practiceapp/controller/post/CreatePostControllerTest.java]: Yes you are right. I updated that.

**Review by irem-zeynep — APPROVED** [review]
Looks good, thanks!

**Pull request merged**